src/main/java/org/polyfill/controllers/PolyfillController.java

```diff
@@ -30,33 +29,40 @@
     UserAgentParserService userAgentParserService;
 
     @Autowired
-    @Qualifier("presort")
-    PolyfillQueryService preSortPolyfillQueryService;
+    PolyfillQueryService polyfillQueryService;
 
     @RequestMapping(value = "/polyfill.{type}", method = RequestMethod.GET)
     public View polyfillApi(@RequestHeader("User-Agent") String uaString,
+                            @RequestParam(value = "features", required = false) String features,
+                            @RequestParam(value = "excludes", required = false) String excludes,
+                            @RequestParam(value = "flags", required = false) String flags,
                             @PathVariable String type, Model model) {
-        if (type.equals("js")) {
-            List<Polyfill> polyfillList = preSortPolyfillQueryService.getPolyfillsByUserAgent(userAgentParserService.parse(uaString));
 
-            return new PolyfillsView(polyfillList, "raw");
+        if (type.equals("js")) {
+            boolean doMinify = false;
+            String polyfillsSource = getPolyfillsSource(uaString, features, excludes, flags, doMinify);
+            return new PolyfillsView(polyfillsSource, doMinify);
```

Comment:
I was having trouble to do all the source concatenation in PolyfillsView. Right now getPolyfillsSource returns the entire polyfill source, so it needs minify there to determine whether to get raw/min source.
PolyfillsView contains code for generating the closure source and comments, so it needs doMinify.

1. One way to move all the source concatenating logic to PolyfillsView is to also attach a polyfill object to FeatureOptions. This way we would be able to access the source and flags in PolyfillsView.
2. Another way to do this would be create a new service for source concatenation, closure, and comments.

What do you think? Also, should we do this in a refactor commit or this commit?